File: backend/app/modules/users/models/user_model.py
"""Internal domain model for a user document."""
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, EmailStr, Field

from app.core.constants import UserRole


class UserModel(BaseModel):
    """Shape of a user document stored in MongoDB."""

    email: EmailStr
    username: str
    hashed_password: str
    full_name: str
    role: UserRole = UserRole.USER
    is_active: bool = True
    is_email_verified: bool = False
    avatar_url: Optional[str] = None
    default_currency: str = "USD"
    email_verify_token: Optional[str] = None
    password_reset_token: Optional[str] = None
    password_reset_expires: Optional[datetime] = None
    last_login_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

File: frontend/src/app/core/models/user.model.ts
export interface User {
  id: string;
  email: string;
  username: string;
  full_name: string;
  role: 'user' | 'admin';
  is_active: boolean;
  is_email_verified: boolean;
  avatar_url?: string | null;
  default_currency: string;
  last_login_at?: string | null;
  created_at?: string | null;
}

export interface TokenPair {
  access_token: string;
  refresh_token: string;
  token_type: string;
}

export interface AuthResult {
  user: User;
  tokens: TokenPair;
}

export interface RegisterPayload {
  full_name: string;
  username: string;
  email: string;
  password: string;
}

export interface LoginPayload {
  identifier: string;
  password: string;
}

export interface ProfileUpdate {
  full_name?: string;
  avatar_url?: string | null;
  default_currency?: string;
}

export interface PasswordChange {
  current_password: string;
  new_password: string;
}
